src/math/MultiplierPointMath.sol

```diff
@@ -0,0 +1,166 @@
+     * @param _balance quantity of tokens
+     * @return maxMPAccrued maximum quantity of muliplier points that can be generated for given `_balance`
+     */
+    function _maxAccrueMP(uint256 _balance) internal pure returns (uint256 maxMPAccrued) {
+        return Math.mulDiv(_balance, MP_MPY, 100);
+    }
+
+    /**
+     * @notice The maximum total multiplier points that can be generated for a determined amount of balance and lock
+     * duration.
+     * @param _balance Represents the current account balance
+     * @param _lockTime The time duration for which the balance is locked
+     * @return maxMP Maximum Multiplier Points that can be generated for given `_balance` and `_lockTime`
+     */
+    function _maxTotalMP(uint256 _balance, uint256 _lockTime) internal pure returns (uint256 maxMP) {
+        return _balance + Math.mulDiv(_balance * MP_APY, (MAX_MULTIPLIER * YEAR) + _lockTime, YEAR * 100);
+    }
+
+    /**
+     * @notice The absolute maximum total multiplier points that some balance could have, which is the sum of the
+     * maximum
+     * lockup time bonus possible and the maximum accrued multiplier points.
+     * @param _balance quantity of tokens
+     * @return maxMPAbsolute Absolute Maximum Multiplier Points
+     */
+    function _maxAbsoluteTotalMP(uint256 _balance) internal pure returns (uint256 maxMPAbsolute) {
+        return Math.mulDiv(_balance, MP_MPY_ABSOLUTE, 100);
+    }
+
+    /**
+     * @dev Caution: This value is estimated and can be incorrect due precision loss.
+     * @notice Calculates the remaining lock time available for a given `_mpMax` and `_balance`
+     * @param _balance Current balance used to calculate the maximum multiplier points.
+     * @param _mpMax Maximum multiplier points calculated from the current balance.
+     * @return lockTime Amount of lock time allowed to be increased
+     */
+    function _lockTimeAvailable(uint256 _balance, uint256 _mpMax) internal pure returns (uint256 lockTime) {
+        return Math.mulDiv((_balance * MP_MPY_ABSOLUTE) - _mpMax, YEAR, _balance * 100);
+    }
+
+    /**
+     * @notice Calculates the time required to accrue a specific multiplier point value.
+     * @param _balance The current balance.
+     * @param _mp The target multiplier points to accrue.
+     * @return timeToReachMaxMP The time required to reach the specified multiplier points, in seconds.
+     */
+    function _timeToAccrueMP(uint256 _balance, uint256 _mp) internal pure returns (uint256 timeToReachMaxMP) {
+        return Math.mulDiv(_mp * 100, YEAR, _balance * MP_APY);
+    }
+
+    /**
+     * @notice Calculates the bonus multiplier points based on the balance and maximum multiplier points.
+     * @param _balance The current balance.
+     * @param _maxMP The maximum multiplier points.
+     * @return bonusMP The calculated bonus multiplier points.
+     */
+    function _retrieveBonusMP(uint256 _balance, uint256 _maxMP) internal pure returns (uint256 bonusMP) {
```

Comment:
This is not used anywhere. Where is this supposed to go?

Reply:
MultiplierPointMath.sol, similar to other library contracts, such as SafeMath.sol, may have functions that are not used, but are available to facilitate the development.

Currently this function is not used, however, it can be used to implement a view function which could separate the constituents of total MP.

Reply:
Okay, fair enough.